book/src/engine/logic.md

```diff
@@ -0,0 +1,123 @@
+# Logic
+
+## Overview
+
+`chalk-engine` solves a `Goal` using a depth-first search. When asked to solve a
+particular `Goal` it hasn't seen before, it will first ask the [`Context`] to
+generate a set of program clauses, that get turned into `Strand`s, that could
```

Comment:
I suggest we link all data structures to the chalk rustdocs if we can

Reply:
Ok, I've tried to be much more liberal with my use of links.